shared_memory_dict/dict.py

```diff
@@ -54,15 +74,71 @@ def __delitem__(self, key: str) -> None:
         with self._modify_db() as db:
             del db[key]
 
+    def __iter__(self) -> Iterator:
+        return iter(self._read_memory())
+
+    def __reversed__(self):
+        return reversed(self._read_memory())
+
     def __del__(self) -> None:
         self.cleanup()
 
-    def __contains__(self, key: object) -> bool:
+    def __contains__(self, key: str) -> bool:
         return key in self._read_memory()
 
+    def __eq__(self, other: Any) -> bool:
+        return self._read_memory() == other
+
+    def __ne__(self, other: Any) -> bool:
+        return self._read_memory() != other
+
+    if sys.version_info > (3, 8):
+
+        def __or__(self, other: Any):
+            return self._read_memory() | other
+
+        def __ror__(self, other: Any):
+            return other | self._read_memory()
+
+        def __ior__(self, other: Any):
+            with self._modify_db() as db:
+                db |= other
+                return db
+
+    def __str__(self) -> str:
+        return str(self._read_memory())
+
+    def __repr__(self) -> str:
+        return repr(self._read_memory())
+
+    def get(self, key: str, default: Optional[Any] = None) -> Any:
+        return self._read_memory().get(key, default)
+
     def keys(self) -> KeysView[Any]:  # type: ignore
         return self._read_memory().keys()
 
+    def values(self) -> ValuesView[Any]:  # type: ignore
+        return self._read_memory().values()
+
+    def items(self) -> ItemsView:  # type: ignore
+        return self._read_memory().items()
+
+    __marker = object()
```

Comment:
what about change this control const to a const on top of the module with the name `SENTINAL`?, f.ex:
```python
_SENTINEL = object()
...
def pop(self, key: str, default: Optional[Any] = SENTINEL):
    with self._modify_db() as db:
        if default is SENTINEL:
            return db.pop(key)
        return db.pop(key, default)
```
Just to make this code more explicit, what do you think about it?

Reply:
Named it based on [default MuttableMaping](https://github.com/python/cpython/blob/73607be68668ab7f4bee53507c8dc7b5a46c9cb4/Lib/_collections_abc.py#L900) but I agree with that change, sentinel is a better name.